tools/devops/device-tests/templates/device-tests.yml

```diff
@@ -0,0 +1,303 @@
+    echo "#$X[task.addattachment type=Distributedtask.Core.Summary;name=Html Report;]$PWD/HtmlReportBoston.md"
+    set -x
+
+    echo "Html Report: http://xamarin-storage/$P/jenkins-results/tests/index.html"
+  displayName: 'Print Html Report link'
+  condition: and(always(), eq('${{ parameters.htmlReportStorage}}', 'xamarin-storage')) # execute always but when we are using vsdrops since we want to send the message and the url at the same time
+
+# Run tests. If we are using xamarin-storage add a periodic command to be executed by xharness, else, since we are using vsdrops do nothing.
+- bash: |
+    set -x
+    set -e
+
+    cd $WORKING_DIR/xamarin-macios
+    make -C builds downloads -j || true
+    make -C builds .stamp-mono-ios-sdk-destdir -j || true
+    if [[ "$HTML_STORAGE" == "xamarin-storage" ]]; then
+       export TESTS_PERIODIC_COMMAND="--periodic-interval 10 --periodic-command rsync --periodic-command-arguments '-avz -e \"ssh\" $PWD/jenkins-results builder@xamarin-storage:/volume1/storage/$XAMARIN_STORAGE_PATH'"
+    fi
+
+    MONO_ENV_OPTIONS=--trace=E:all make -C tests vsts-device-tests
+  env:
+    WORKING_DIR: $(System.DefaultWorkingDirectory) 
+    XAMARIN_STORAGE_PATH: $(XAMARIN_STORAGE_PATH)
+    TESTS_EXTRA_ARGUMENTS: ${{ parameters.testsLabels }}
+    HTML_STORAGE: ${{ parameters.htmlReportStorage }}
+  displayName: 'Run tests'
+  timeoutInMinutes: 600
+
+# Upload TestSummary as an artifact.
+- task: PublishPipelineArtifact@1
+  displayName: 'Publish Artifact: TestSummary'
+  inputs:
+    targetPath: 'xamarin-macios/tests/TestSummary.md'
+    artifactName: TestSummary.md
+  continueOnError: true
+  condition: succeededOrFailed()
+
+# Use the cmdlet to post a new summary comment. The cmdlet checks if we have the TestSummary.md file or not. It will also add the appropiate links to the commnet. 
+# This step is ONLY executed when we selected the xamarin-storage option. If we use vsdrops we want to wait until the publish_html job to do it so that we can 
+# calculate the url of the html report based in the vsdrop path.
+- pwsh: |
+    Import-Module $Env:SYSTEM_DEFAULTWORKINGDIRECTORY/xamarin-macios/tools/devops/device-tests/scripts/GitHub.psm1 
+    New-GitHubSummaryComment -Context "$Env:CONTEXT" -XamarinStoragePath "$Env:XAMARIN_STORAGE_PATH" -TestSummaryPath "$Env:SYSTEM_DEFAULTWORKINGDIRECTORY/xamarin/xamarin-macios/tests/TestSummary.md"
+  env:
+    BUILD_REVISION: $(BUILD_REVISION)
+    CONTEXT: ${{ parameters.statusContext }}
+    GITHUB_TOKEN: $(GitHub.Token)
+    XAMARIN_STORAGE_PATH: $(XAMARIN_STORAGE_PATH) # could be null if the step was not executed if we use vsdrops, the cmdlet will know how to handle that case
+  displayName: 'Add summaries'
+  condition: and(always(), eq('${{ parameters.htmlReportStorage}}', 'xamarin-storage')) # execute always but when we are using vsdrops since we want to send the message and the url at the same time from the pusblish_hml job
```

Comment:
```suggestion
  condition: and(always(), eq('${{ parameters.htmlReportStorage }}', 'xamarin-storage')) # execute always but when we are using vsdrops since we want to send the message and the url at the same time from the pusblish_hml job
```